src/scf/self_consistent_field.jl

```diff
@@ -69,73 +71,98 @@ Solve the Kohn-Sham equations with a SCF algorithm, starting at ρ.
     occupation = nothing
     eigenvalues = nothing
     ρout = ρ
+    ρspinout = ρspin
     εF = nothing
     n_iter = 0
     energies = nothing
     ham = nothing
-    info = (n_iter=0, ρin=ρ)   # Populate info with initial values
+    info = (n_iter=0, ρin=ρ, ρspinin=ρspin)   # Populate info with initial values
+    converged = false
 
     # We do density mixing in the real representation
     # TODO support other mixing types
     function fixpoint_map(x)
+        converged && return x  # No more iterations if convergence flagged
+
         n_iter += 1
-        ρin = from_real(basis, x)
+        if n_spin == 2  # TODO So ugly
+            # x has 2 blocks: total and spin density
+            ρin, ρspinin = from_real(basis,x[:, :, :, 1]), from_real(basis,x[:, :, :, 2])
+        else
+            ρin, ρspinin = from_real(basis, x), nothing
+        end
```

Comment:
For this kind of things we should have (here and at other places) functions `pack`/`unpack` that do this

Reply:
Yes. I did not do this because it should go once we merge ρ and ρspin.